test/functional/lua/list_spec.lua

```diff
@@ -0,0 +1,65 @@
+-- Test suite for testing interactions with API bindings
```

Comment:
thanks for creating list_spec 👍 
```suggestion
-- Test suite for vim.list
```